### ISSUE TITLE
move rights/access normalization code out of bin/validate-rights and into a normalization service class for rights

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/dor-services-app#2787]
It will be best to have a consistent approach for "normalizing" our data transforms for any and all fedora datastreams -- that is, accommodating the pre-approved differences between the original xml in a fedora datastream and the roundtrip datastream.

descMetadata / mods is already doing something like this, but we will need to add this for

- rightsMetadata
- identityMetadata
- administrativeMetadata (APOs only)
... (probably all datastreams, but :shrug: 

rightsMetadata currently has such normalization code baked into the bin/xxx-rights-xxx  script -- this code should MOVE to app/services/cocina/normalization/xxxx and the script should be updated there.

I suggest talking to me or JLitt before starting on this ticket as it might make the work clearer.

Note that rightsMetadata ALSO needs to properly normalize the license/use changes (e.g from "by-nd" in original xml to URI) rather than strip elements and ignore them.  (What if we have additional use elements ?   stripping them would prevent us from knowing this)

So this ticket has 3 pieces (3 PRs?  or combine them as you see fit):
- [x] refactor descMetadata normalization code to slightly new approach (or not) so all is under app/services/cocina/normalizion (and make sure bin/xxx scripts work with this change, too)
- [x] create rightsMetadata normalization code in consistent way to mods normalizations, pulling it out of bin/**rights*** roundtrippings script and calling the newly located code properly, as bin/**descMetadata** does
- [ ] fix the rightsMetadata licensing normalization code to o properly normalize the license/use changes (e.g from "by-nd" in original xml to URI) rather than strip elements and ignore them.

**Comment by ndushay** [comment]
after this is done, ping Naomi to add the rights normalization to the rights/access roundtrip unit tests

**Comment by peetucket** [comment]
Part 1 is done (though I used the folder name "normalizers") in https://github.com/sul-dlss/dor-services-app/pull/2790

There is test coverage for that code already, but it would be nice to test the branch...is deploying to qa and running a few druids through a good way to do this?

**Comment by ndushay** [comment]
yay!   i think to test the bin/validate type scripts, you only need to test on sdr-deploy.  that is where the cache lives, so its the only place we run that code.

**Comment by peetucket** [comment]
Part 2 is now done: #2791

**Comment by peetucket** [comment]
I think part 3 in your list is actually in this ticket #2709, so I think we can now close this one

**Comment by ndushay** [comment]
I'm fine with this ticket being closed, but #2709 is actually about APO `defaultObjectRights.xml` ... which looks just like a rightsMetadata.xml, but is only on APO objects and used to populate rightsMetadata.xml sometimes when items are created.   So I made a new ticket #2797 for the remaining work here, and linked it to #2709.  :-)

**Comment by peetucket** [comment]
head explode emoji here
